overlord/configstate/config/transaction.go

```diff
@@ -46,6 +46,23 @@ type Transaction struct {
 	changes  map[string]map[string]interface{}
 }
 
+// RunTransaction holds a transaction with a task that is in charge of
+// appliying a change to the configuration. It is used in the context of
+// configcore.
+type RunTransaction struct {
```

Comment:
a different approach to this would be do add Cache/Cached methods to Transaction like state, and stick the task on the Transaction that way